### DISCUSSION TITLE
Support commit amend

### DISCUSSION THREAD

**Discussion by Andy-AO** [extrawurst/gitui#1011] [Q&A] [answered]
Sometimes I want to use `git commit --amend --no-edit` to append files that have been staged to a previous commit, which happens quite often, because I often find that there is a small problem that needs to be changed.

My current method is to exit gitui and then use PowerShell, but this is too cumbersome.

Is it possible to customize the action of a specific git command or executable?

That way it's just a matter of hitting the keyboard to do these personalized actions.

**Comment by extrawurst** [comment, marked as answer]
Maybe I am missing something here but once you are in the commit popup you should see the amend command available. Does that give you what you need?

![image](https://user-images.githubusercontent.com/776816/143542115-cfc00cde-c49c-485f-9b85-8c013e44db8d.gif)

**Reply by Andy-AO** [reply]
Yes, that's what I want.Thank you. I didn't see change down there when this commit window opened.

**Reply by mloskot** [reply]
@extrawurst Apologies if I'm making a silly mistake (gitui newbie here), but has this `Amend` command been moved?
I can not find it anywhere in 0.22.1

I can only see `Edit [e]` in Log > Inspect, but it does not seem to do anything, no commit edit pops up

![image](https://github.com/extrawurst/gitui/assets/80741/51cb1c84-9d3b-4454-8acc-636eafebabf4)

**Reply by willtonkin** [reply]
hey @mloskot, I believe the amend option is only available from the `status [1]` view when you're preparing a new commit, not from the `log [2]` view.

As it is analogous to performing a `git commit --amend` which allows you to add your current changes to the previous commit and amend the commit message.

**Reply by mloskot** [reply]
@willtonkin I think I get it now. The `Amend` is available while in ongoing `Commit` action. It makes sense.

I kind of expected it to work like in GUI https://fork.dev which allows me to amend last commit even if there are no staged changes.
In gitui, it is also possible but the sequence is different, goes via Unstaging, and also gitui seems losing message of that commit being amened.

**Reply by sainttttt** [reply]
@mloskot So I had this same question, and the way I think you can do it is do a soft reset which will stage your changes, and then for the commit message you can press `Control-N` which will go through the history of your prior commit messages. You can pick the last one and then edit it. Also if you hit `Control-N` too many times and overshoot it, just hit `Esc` and cancel the commit dialog and reopen it and then start hitting `Control-N` again and it will start populating commit messages from the top again. Hope this works and if you figured out a better system let me know! (Or if this doesn't work)

**Reply by mloskot** [reply]
@sainttttt Thanks a lot for chiming in and sharing the solution. 

> Hope this works and if you figured out a better system let me know! (Or if this doesn't work)

I don't use GitUI. I use the other Git TUI ;) But, I may give GitUI another go at some point. I'll report back if I do.
